lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,399 @@
+redhat_repositories:
+    description: list of repositories and their data
+    returned: always
+    type: dict
+    sample: '{
+        "redhat_repositories": {
+            "rhel_7_server_extras_rpms": {
+                "enabled": true,
+                "id": "rhel-7-server-extras-rpms",
+                "name": "Red Hat Enterprise Linux 7 Server - Extras (RPMs)",
+                "url": "https://cdn.redhat.com/content/dist/rhel/server/7/7Server/$basearch/extras/os"
+            },
+            "rhel_7_server_rpms": {
+                "enabled": true,
+                "id": "rhel-7-server-rpms",
+                "name": "Red Hat Enterprise Linux 7 Server (RPMs)",
+                "url": "https://cdn.redhat.com/content/dist/rhel/server/7/$releasever/$basearch/os"
+            }
+        }
+    }'
+
+'''
+
+import os
+import re
+import types
+import ConfigParser
+import shlex
+import syslog
+import pprint
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+
+debug=False
+
+def notice(msg):
+    if debug:
+        syslog.syslog(syslog.LOG_NOTICE, msg)
+
+if debug:
+    syslog.openlog('ansible-%s' % os.path.basename(__file__))
+
+class RhsmRepository(object):
+    '''
+        This class to host Repository information and enable/disable per repo
+    '''
+    def __init__(self, module, **kwargs):
+        self.module = module
+        for k,v in kwargs.items():
+            setattr(self, k, v)
+
+    def __str__(self):
+        return str(self.__getattribute('_name'))
+
+    @property
+    def id(self):
+        return self.RepoID
+
+    @property
+    def name(self):
+        return self.RepoName
+
+    @property
+    def url(self):
+        return self.RepoURL
+
+
+    @property
+    def is_enabled(self):
+        return self.Enabled
+
+    @property
+    def is_disabled(self):
+        return not self.Enabled
+
+class RhsmRepositories(object):
+    '''
+        This class to manipulate repositories
+    '''
+
+    def __init__(self, module):
+       self.module = module
+       self.repos = self._load_repo_list()
+
+    def __iter__(self):
+        return self.repos.__iter__()
+
+    def _load_repo_list(self):
+        """
+            Loads list of all available repos, whether enabled or not
+        """
+
+        args = "subscription-manager repos --list"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True, environ_update=dict(LANG='C'))
+
+        repos = []
+        for line in stdout.split('\n'):
+
+            # Remove leading+trailing whitespace
+            line = line.strip()
+
+            # An empty line implies the end of a output group, strip headers too
+            if len(line) == 0 or re.match(r'^\s+.*', line) or re.match(r'^\+s.*', line):
+                continue
+
+                key = ''
+                value = ''
+                continue
+
+            # If a colon ':' is found, parse
+            elif ':' in line:
+                (key, value) = line.split(':',1)
```

Comment:
Missing space in front of `1`. I strongly recommend to use `flake8` tool on your code before you submit it. It significantly improve the code formatting which has direct influence on the readability of the code.